src/config/field-overrides/program-indicator/OrgUnitField.js

```diff
@@ -0,0 +1,323 @@
+  d2 model. Since these are mutable objects the values we find in `prevProps.model`
+  are identical to the ones in `this.props.model`. To address this issue we have to
+  keep track of the previous values manually
+*/
+
+export default class OrgUnitField extends Component {
+    constructor(props, context) {
+        super(props);
+        this.d2 = context.d2;
+        this.translate = this.d2.i18n.getTranslation.bind(context.d2.i18n);
+        this.state = {
+            loading: false,
+            error: null,
+            value: undefined,
+            dataElements: {},
+        };
+        this.staticOptions = {
+            event: {
+                value: ANALYTICS_TYPE_EVENT,
+                text: this.translate('Event'),
+            },
+            enrollment: {
+                value: ANALYTICS_TYPE_ENROLLMENT,
+                text: this.translate('Enrollment'),
+            },
+            ownerAtStart: {
+                value: 'OWNER_AT_START',
+                text: this.translate('Owner at start'),
+            },
+            ownerAtEnd: {
+                value: 'OWNER_AT_END',
+                text: this.translate('Owner at end'),
+            },
+            registration: {
+                value: 'REGISTRATION',
+                text: this.translate('Registration'),
+            },
+        };
+        this.onChange = this.onChange.bind(this);
+
+        this.updateManualPrevProps(props);
+    }
+
+    componentDidMount() {
+        if (this.hasRequiredParams()) {
+            this.fetchDataElements();
+        }
+
+        this.updateManualPrevProps();
+    }
+
+    componentDidUpdate() {
+        if (
+            this.hasRelevantPropChanged() &&
+            this.shouldFetchDataElementsForProgram()
+        ) {
+            this.fetchDataElements();
+        }
+
+        this.updateManualPrevProps();
+    }
+
+    updateManualPrevProps(props = this.props) {
+        this.manualPrevProps = {
+            programId: props.model.program.id,
+            analyticsType: props.model.analyticsType,
+        };
+    }
+
+    hasRelevantPropChanged() {
+        return this.hasProgramChanged() || this.hasAnalyticsTypeChanged();
+    }
+
+    hasProgramChanged() {
+        return this.manualPrevProps.programId !== this.props.model.program.id;
+    }
+
+    hasAnalyticsTypeChanged() {
+        return (
+            this.manualPrevProps.analyticsType !==
+            this.props.model.analyticsType
+        );
+    }
+
+    async fetchDataElements() {
+        this.setState({
+            loading: true,
+            error: null,
+        });
+
+        try {
+            const api = this.d2.Api.getApi();
+            const url = `programs/${this.props.model.program.id}`;
+            const data = {
+                fields:
+                    'programStages[id,programStageDataElements[dataElement[id,displayName,valueType]]',
+            };
+            const allDataElements = await api.get(url, data);
```

Comment:
The more "d2" or "maintenance" way to do this would be something like

```js
const allDataElements = await this.d2.models.program.get(this.props.model.program.id, { fields: data.fields})
```

Shouldn't really matter though, but then you don't have to concat the url yourelf.